src/VisualStudio/Core/Def/Implementation/IntellisenseControls/IntellisenseTextBox.cs

```diff
@@ -0,0 +1,513 @@
+                editorOptions.SetOptionValue(DefaultWpfViewOptions.AppearanceCategory, AppearanceCategory);
+            }
+
+            Typeface typeface = new Typeface(container.FontFamily, container.FontStyle, container.FontWeight, container.FontStretch);
+            _classificationFormatMap.DefaultTextProperties = _classificationFormatMap.DefaultTextProperties.SetTypeface(typeface);
+            _classificationFormatMap.DefaultTextProperties = _classificationFormatMap.DefaultTextProperties.SetFontRenderingEmSize(container.FontSize);
+
+            ErrorHandler.ThrowOnFailure(viewModel.VsTextView.AddCommandFilter(this, out this._nextCommandTarget));
+
+            // Get the host control to render the view
+            this._textViewHost = _editorAdapterFactory.GetWpfTextViewHost(viewModel.VsTextView);
+
+            BindingOperations.SetBinding(_textViewHost.HostControl, AutomationProperties.NameProperty, new Binding { Source = container, Path = new PropertyPath(AutomationProperties.NameProperty) });
+            BindingOperations.SetBinding(_textViewHost.HostControl, AutomationProperties.LabeledByProperty, new Binding { Source = container, Path = new PropertyPath(AutomationProperties.LabeledByProperty) });
+
+            // For non-blurry text
+            TextOptions.SetTextFormattingMode(this._textViewHost.HostControl, TextFormattingMode.Display);
+            this._textViewHost.HostControl.Loaded += this.HostControl_Loaded;
+
+            this.AddLogicalChild(this._textViewHost.HostControl);
+            this.AddVisualChild(this._textViewHost.HostControl);
+
+            ContainerName = container.Name;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether there is an active intellisense session.
+        /// </summary>
+        public bool HasActiveIntellisenseSession
+        {
+            get
+            {
+                if (this._textViewHost.TextView.Properties.TryGetProperty(typeof(ICompletionBroker), out ICompletionBroker completionBroker))
+                {
+                    return completionBroker.IsCompletionActive(this._textViewHost.TextView);
+                }
+
+                if (this._textViewHost.TextView.Properties.TryGetProperty(typeof(IIntellisenseSessionStack), out IIntellisenseSessionStack intellisenseSessionStack))
+                {
+                    return intellisenseSessionStack.Sessions.Count > 0;
+                }
+
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Shuts down any active Intellisense session.
+        /// </summary>
+        public void ShutDownIntellisenseSessions()
+        {
+            if (this._textViewHost.TextView.Properties.TryGetProperty(typeof(ICompletionBroker), out ICompletionBroker completionBroker))
+            {
+                completionBroker.DismissAllSessions(this._textViewHost.TextView);
+            }
+        }
```

Comment:
Is there some better way to disable this with an editor property or something? Playing whack-a-mole like this seems questionable.

Reply:
Does this also have any issues around async completion?

Reply:
Filed https://github.com/dotnet/roslyn/issues/41830